### PR TITLE
Check if option is url for pip

### PR DESCRIPTION
**Pull request by SamYaple**
Requirements and Constraints file can be a local file or a url. Add
a simple check to use the appropriate path

LP: #1621543

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: snapcraft/plugins/python.py [hunk 1]
```diff
@@ -205,11 +210,7 @@ def _run_pip(self, setup):
             self.run(pip_install + ['.'], cwd=cwd, env=env)
 
     def _fix_permissions(self):
-        for root, dirs, files in os.walk(self.installdir):
-            for filename in files:
-                _replicate_owner_mode(os.path.join(root, filename))
-            for dirname in dirs:
-                _replicate_owner_mode(os.path.join(root, dirname))
```
Comment (SamYaple) [inline snapcraft/plugins/python.py]: not sure why this got removed. probably bad rebase. will address

File: snapcraft/plugins/python.py [hunk 2]
```diff
@@ -252,3 +253,6 @@ def _replicate_owner_mode(path):
     if file_mode & stat.S_IRUSR:
         new_mode |= stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH
     os.chmod(path, new_mode)
+
+def is_str_url(s):
```
Comment (sergiusens) [inline snapcraft/plugins/python.py]: instead of this `from snapcraft.common import isurl` and use that

File: snapcraft/plugins/python.py [hunk 3]
```diff
@@ -192,10 +195,12 @@ def _run_pip(self, setup):
         env = self._get_build_env()
 
         if self.options.requirements:
-            self.run(pip_install + [
-                '--requirement',
-                os.path.join(self.sourcedir, self.options.requirements),
-            ], env=env)
+            if is_str_url(self.options.requirements):
```
Comment (sergiusens) [inline snapcraft/plugins/python.py]: instead of this `from snapcraft.common import isurl` and use that

File: snapcraft/plugins/python.py [hunk 4]
```diff
@@ -165,10 +166,12 @@ def _get_pip_command(self):
                        '--disable-pip-version-check']
 
         if self.options.constraints:
-            pip_install = pip_install + [
-                '--constraint',
-                os.path.join(self.sourcedir, self.options.constraints),
-            ]
+            if is_str_url(self.options.constraints):
```
Comment (sergiusens) [inline snapcraft/plugins/python.py]: instead of this `from snapcraft.common import isurl` and use that

File: snapcraft/plugins/python.py [hunk 5]
```diff
@@ -45,11 +45,12 @@
       The python version to use. Valid options are: python2 and python3
 """
 
+from glob import glob
 import os
 import re
 import stat
 import subprocess
-from glob import glob
+import urlparse
```
Comment (sergiusens) [inline snapcraft/plugins/python.py]: instead of this `from snapcraft.common import isurl` and use that

File: snapcraft/plugins/python.py (hunk 2 above)
Comment (SamYaple) [inline snapcraft/plugins/python.py]: ah nice. already a built in. will replace it

File: snapcraft/plugins/python.py [hunk 6]
```diff
@@ -45,14 +45,15 @@
       The python version to use. Valid options are: python2 and python3
 """
 
+from glob import glob
 import os
 import re
 import stat
 import subprocess
-from glob import glob
```
Comment (sergiusens) [inline snapcraft/plugins/python.py]: why was this swapped to the top?

**Comment by sergiusens** [comment]
ok to test

**Comment by sergiusens** [comment]
Looks good except for the unnecessary move of the glob import

File: snapcraft/plugins/python.py (hunk 6 above)
Comment (SamYaple) [inline snapcraft/plugins/python.py]: this was not intentional. not sure why it was moved. fixing

**Pull request merged**